File: Trees/Binary Tree Zigzag Level Order Traversal/recur.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode * root, vector<vector<int>> & v, int level){
        if (root==NULL) return;
        else if (level>=v.size()) {
            vector<int> l;
            l.emplace_back(root->val);
            v.emplace_back(l);
        }
        else if (level%2!=0)  {
            v[level].insert(v[level].begin(),root->val);
        }
        else v[level].emplace_back(root->val);
        helper(root->left,v,level+1);
        helper(root->right,v,level+1);
        return;
    }
    vector<vector<int>> zigzagLevelOrder(TreeNode* root) {
        vector<vector<int>> v;
        helper(root,v,0);
        return v;
    }
};

File: Trees&Tries/Binary Tree Level Order Traversal/recur.py
https://leetcode.com/problems/binary-tree-level-order-traversal/
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        lst=[]
        def recur(root,level):
            nonlocal lst
            if root==None: return
            if level==len(lst): lst.append([root.val]) 
            else: lst[level].append(root.val)
            recur(root.left,level+1)
            recur(root.right,level+1)            
        recur(root,0)
        return lst